src/org/sosy_lab/java_smt/test/ModelTest.java

```diff
@@ -2254,6 +2259,453 @@ public void arrayTest5()
+  private Iterable<Formula> getFunctionArgs(Formula f) {
+    requireVisitor();
+    return mgr.visit(
+        f,
+        new DefaultFormulaVisitor<>() {
+          @Override
+          protected Iterable<Formula> visitDefault(Formula f) {
+            throw new IllegalArgumentException();
+          }
+
+          @Override
+          public Iterable<Formula> visitFreeVariable(Formula f, String name) {
+            // Base case for array values
+            return ImmutableList.of();
+          }
+
+          @Override
+          public Iterable<Formula> visitFunction(
+              Formula f, List<Formula> args, FunctionDeclaration<?> functionDeclaration) {
+            if (functionDeclaration.getKind().equals(FunctionDeclarationKind.SELECT)) {
+              return FluentIterable.concat(
+                  getFunctionArgs(args.get(0)), ImmutableList.of(args.get(1)));
+            } else {
+              // Assume its an UF value
+              return args;
+            }
+          }
+        });
+  }
+
+  /** Evaluate the UF or array constraint and check that the model contains the expected values. */
+  private <T> void checkModelContains(
+      BooleanFormula pFormula, String pVar, Map<List<Formula>, Formula> expected) {
+    try (var prover = context.newProverEnvironment(ProverOptions.GENERATE_MODELS)) {
+      prover.addConstraint(pFormula);
+      assertThat(prover.isUnsat()).isFalse();
+
+      ImmutableMap.Builder<List<Formula>, Formula> builder = ImmutableMap.builder();
+      for (var assignment : prover.getModel().asList()) {
+        if (assignment.getName().equals(pVar)) {
+          builder.put(
+              ImmutableList.copyOf(getFunctionArgs(assignment.getKey())),
+              assignment.getValueAsFormula());
+        }
+      }
+      var model = builder.build();
+      assertThat(model).containsAtLeastEntriesIn(expected);
+
+    } catch (SolverException | InterruptedException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Test
+  public void testArray1Bv() {
+    requireArrays();
+    requireBitvectors();
+
+    var bitvectorType = FormulaType.getBitvectorTypeWithSize(8);
+    var array = amgr.makeArray("array", bitvectorType, bitvectorType);
+
+    var idx = bvmgr.makeBitvector(8, 1);
+    var val = bvmgr.makeBitvector(8, 10);
+
+    checkModelContains(
+        bvmgr.equal(amgr.select(array, idx), val),
+        "array",
+        ImmutableMap.of(ImmutableList.of(idx), val));
+  }
+
+  @Test
+  public void testArray2Bv() {
+    requireArrays();
+    requireBitvectors();
+
+    var bitvectorType = FormulaType.getBitvectorTypeWithSize(8);
+    var array = amgr.makeArray("array", bitvectorType, bitvectorType);
+
+    var idxA = bvmgr.makeBitvector(8, 1);
+    var valA = bvmgr.makeBitvector(8, 10);
+
+    var idxB = bvmgr.makeBitvector(8, 2);
+    var valB = bvmgr.makeBitvector(8, 5);
+
+    checkModelContains(
+        bmgr.and(
+            bvmgr.equal(amgr.select(array, idxA), valA),
```

Comment:
Is this not covered by existing tests?
(It might be that there are only Integer based tests for this)